Controller/Api/ContentController.php

```diff
@@ -24,14 +24,35 @@ class ContentController extends Controller
      */
     public function indexAction(Request $request)
     {
+        return $this->retrieveContent($request);
+    }
+
+    /**
+     * Archive
+     *
+     * @param Request $request
+     *
+     * @return null|JsonResponse|Response
+     */
+    public function archiveAction(Request $request)
+    {
+        return $this->retrieveContent($request, true);
+    }
+
+    /**
+     * @param Request $request
+     * @param bool $archive
+     * @return null|JsonResponse|Response
+     */
+    public function retrieveContent(Request $request, $archive = false)
```

Comment:
Could you move this to a separate service? Maybe inside the content manager?
Controllers should only have `Actions`.